### DISCUSSION TITLE
Meshlet size

### DISCUSSION THREAD

**Discussion by ib00** [zeux/meshoptimizer#679] [Q&A] [answered]
The number of vertices in a meshlet is limited. Is there a way to increase the limit or to specify the maximum number
of vertices?

**Comment by zeux** [comment]
Limited how? The 255 vertex limit is aligned with all existing graphics APIs and hardware from what I know.

**Comment by ib00** [comment]
Yes, of course. My intended use of meshoptimizer was to use it for non-GPU rendering where I need to break the mesh up
into groups of 2K-8K triangles.

**Reply by zeux** [reply, marked as answer]
I would probably try using max supported limits (512 triangles/255 vertices) and merge consecutive meshlets, with or without subsequent vertex deduplication based on whether that’s necessary. It’s not ideal but meshlets are roughly grouped based on proximity so the output should be reasonable.
